### ISSUE TITLE
OSError: -2 when decoding a tiff_lzw file

### ISSUE THREAD

**Issue by kmilos** [python-pillow/Pillow#4237]
This is with pillow 6.2.1 on 

Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 22:01:29) [MSC v.1900 64 bit (AMD64)]

Seems libtiff is found without a problem, but decoding fails:

```python
In [5]: dat = img.getdata()
have fileno, calling fileno version of the decoder.
Traceback (most recent call last):

  File "<ipython-input-5-d82315236633>", line 1, in <module>
    dat = img.getdata()

  File "C:\Users\kmilos\.conda\envs\dev\lib\site-packages\PIL\Image.py", line 1304, in getdata
    self.load()

  File "C:\Users\kmilos\.conda\envs\dev\lib\site-packages\PIL\TiffImagePlugin.py", line 1097, in load
    return self._load_libtiff()

  File "C:\Users\kmilos\.conda\envs\dev\lib\site-packages\PIL\TiffImagePlugin.py", line 1209, in _load_libtiff
    raise IOError(err)

OSError: -2


In [6]: dat = img.getdata()

In [7]: print(dat)
<ImagingCore object at 0x000001FC04A55DB0>
```

Also note that on the second attempt I get an object back, bit if I try to convert to an array, it is filled with zeros, so issue #3863 is not really closed for me.

No problems if I use 'TIFF' or 'TIFF-FI' formats via imageio, but 'TIFF-PIL' fails with the same error.

**Comment by radarhere** [comment]
Could you provide the image that is causing the problem?

**Comment by kmilos** [comment]
Happens also with this tiff_adobe_deflate compressed [mandrill.zip](https://github.com/python-pillow/Pillow/files/3901288/mandrill.zip), so not limited to tiff_lzw...

**Comment by radarhere** [comment]
Loading the mandrill image, I'm unable to replicate using Pillow 6.2.1 with Python 3.7.5 on macOS 10.15.
```python
from PIL import Image
im = Image.open('mandrill.tif')
im.load()
```

**Comment by kmilos** [comment]
On Windows it is line 1189 of tiffimageplugin.py where the problem occurs

`n, err = decoder.decode(b"fpfp")`

results in n=-1 and err=-2

**Comment by radarhere** [comment]
I'm also unable to replicate with 6.2.1 on AppVeyor Windows or master on GHA Windows.

**Comment by kmilos** [comment]
I'm now suspecting this might be due to Anaconda building against new libtiff 4.1.0... Your test is still w/ 4.0.10?

**Comment by radarhere** [comment]
Hmm, unfortunately no, I have now tried Pillow 6.2.1 on AppVeyor Windows with libtiff 4.1.0, and I am still unable to reproduce.

**Comment by kmilos** [comment]
I have reproduced on a second Windows x64 machine, also set up with Anaconda... I reported the issue for the conda-forge package as https://github.com/conda-forge/pillow-feedstock/issues/68

**Comment by kmilos** [comment]
FWIW, I drilled down to 'IMAGING_CODEC_BROKEN` being returned from [ImagingLibTiffDecode()](https://github.com/python-pillow/Pillow/blob/995c863200520500161e81b0ca24c80a99973b81/src/libImaging/TiffDecode.c#L281) but no idea why for now

**Comment by cmlyldz** [comment]
Hello, I have same problem also. 

```
  File "D:\ProgramData\Anaconda3\envs\spyder\lib\site-packages\PIL\TiffImagePlugin.py", line 1209, in _load_libtiff
    raise IOError(err)

OSError: -2
```
Do you find any solution?

**Comment by hugovk** [comment]
https://github.com/conda-forge/pillow-feedstock/issues/69#issuecomment-564052801 has a workaround:

> I can reproduce that on Windows. Downgrading libtiff to 4.0.10 (`conda install libtiff=4.0.10`) works. Investigating it...
> 
> PS: it works fine on macOS and Linux, only Windows is failing.

**Comment by cgohlke** [comment]
It looks like this issue is due to libtiff-4.1.0 CMake builds on Windows now unconditionally enabling `USE_WIN32_FILEIO`:

https://gitlab.com/libtiff/libtiff/commit/59eb35172f8dc9a46469ad0e4877f9c88ab796cd

I built Pillow-git head against a CMake static build of libtiff-4.1.0 and get 50 test failures. Disabling `USE_WIN32_FILEIO` in CMakeLists.txt fixes all the Pillow test failures.

**Comment by cgohlke** [comment]
IIUC, the `TIFFFdOpen` functions from `tif_unix.c` and `tif_win32.c` are not compatible. The first needs a fileno from the C runtime (`py_file_object.fileno()`), the latter a Windows file handle (`msvcrt.get_osfhandle(py_file_object.fileno())`)...

**Comment by cgohlke** [comment]
I submitted an issue to the libtiff project: https://gitlab.com/libtiff/libtiff/issues/173

**Comment by kmilos** [comment]
@cgohlke I also found https://gitlab.com/libtiff/libtiff/merge_requests/93

**Comment by cgohlke** [comment]
There's also https://gitlab.com/libtiff/libtiff/issues/2 from 2008, which means the libtiff maintainers are aware of the incompatibilities of the 2 implementations. But the recent change is that CMake builds were switched to an incompatible version without the option to revert or detect which version is used at compile or runtime...

**Comment by hmaarrfk** [comment]
Trying to get windows patched up on conda-forge https://github.com/conda-forge/libtiff-feedstock/pull/49

**Comment by hmaarrfk** [comment]
@cgohlke the patch definitely broke opencv on conda forge

So strange, I don't know opencv does it:


```
conda install -c conda-forge opencv libtiff=4.0 python=3.7 pillow
```

Then:

```
from PIL import Image
im = Image.open('hopper_lzw.tif')
im.load()
import cv2
cv2.imread('hopper_lzw.tif')
```

both work.

Now, install the unpatched 4.1
```
conda install libtiff=4.1=*1
```
PIL fails, opencv works.

Now, install the patched up 4.1
```
conda install libtiff=4.1=*2
```

PIL works, opencv doesn't.

**Comment by cgohlke** [comment]
That patch is not complete! It merely fixes/demonstrates the Pillow failures. There are other places in `tif_win32.c` that would need to be patched for libtiff to work...

My solution for now is to patch tiff-4.1.0 `CMakeLists.txt` in line ~601 not to use `tif_win32.c`:
```
set(win32_io FALSE)
```

**Comment by hmaarrfk** [comment]
thanks, I made a more thorough patch and it seems to be workign with:

* Pillow
* opencv
* tifffile

All previously comipled, most updated versions of the packages available on conda-forge.

I'll likely try to upload the package to my own conda channel mark.harfouche

**Comment by hmaarrfk** [comment]
You should be able to test the packages ending in `_4` for windows when they finishing uploading in maybe something like 10 mins:
https://anaconda.org/mark.harfouche/libtiff/files

**Comment by darienmorrow** [comment]
I have also run into a similar problem on Windows 10, Python 3.7.6, Pillow 7.0.0.
```
from PIL import Image
p = 'test.tif'
img = Image.open(p)
img_copy = img.copy()
```
yields:
```
Traceback (most recent call last):

  File "C:\Users\darien\Desktop\test.py", line 5, in <module>
    img_copy = img.copy()

  File "C:\ProgramData\Anaconda3\lib\site-packages\PIL\Image.py", line 1084, in copy
    self.load()

  File "C:\ProgramData\Anaconda3\lib\site-packages\PIL\TiffImagePlugin.py", line 1070, in load
    return self._load_libtiff()

  File "C:\ProgramData\Anaconda3\lib\site-packages\PIL\TiffImagePlugin.py", line 1182, in _load_libtiff
    raise OSError(err)

OSError: -2
```

Is this issue still being worked on?

[test.zip](https://github.com/python-pillow/Pillow/files/4335607/test.zip)

**Comment by hmaarrfk** [comment]
are you using conda-forge? are you in a position where you can test the package from my channel?

**Comment by kmilos** [comment]
@hmaarrfk 4.1.0-h885aae3_4 seems to work, thanks

**Comment by hmaarrfk** [comment]
Awesome, we just updated conda forge with the same patch

**Comment by darienmorrow** [comment]
@hmaarrfk sorry for the late reply. Conda has repeatedly borked itself on my machine. After a few fresh installs, I have
```
$ conda list libtiff
# packages in environment at C:\ProgramData\Miniconda3:
#
# Name                    Version                   Build  Channel
libtiff                   4.1.0                h56a325e_0
```
_which allows my basic test to pass_. Note, this is not your patched version of libtiff.* 

* I was unable to convince conda to download any conda-forge version of libtiff that was not 4.0.9.

**Comment by hmaarrfk** [comment]
Ok. We'll keep me updated if the patch I made breaks new things on conda forge (maybe on conda forge though).

I'll probably try to backport the patch from cgholke which is slightly different than what I have.

**Comment by kmilos** [comment]
@hmaarrfk @cgohlke A different patch has been just recently [merged upstream](https://gitlab.com/libtiff/libtiff/-/commit/cad76c5b089d44dd0a0e17cc67de9da93c5655af), wondering if addresses this correctly...

**Comment by trippingbillies41** [comment]
I am currently running into the same issue where I can not load a .tif on windows, but the script works on linux. I installed the 4.10, any other suggestions?

**Comment by hmaarrfk** [comment]
@kmilos i think it was merged after 4.1 was released correct? Maybe we can backport the patch to 4.1? I would prefer if upstream just made a new release.

Easier for us to track. We've been having trouble with libtiff at conda-forge.... help is appreciated.

**Comment by kmilos** [comment]
@hmaarrfk Correct. Unfortunately I don't have a build environment at hand to confirm backporting it would fix this problem, so the workaround in 4.1=h885aae3_4 and later is ok for now. Seems there's been quite a bit of activity around libtiff lately so one can only hope there will be a new release in the not too distant future...

**Comment by hmaarrfk** [comment]
i wouldn't necessarily pin so hard, but you could maybe require a specific build version? though that is more fragile.

Even if i did come out with build #5, then it wouldn't necessarily fix things for you.... you would have to dependo n build 5. :/

**Comment by trippingbillies41** [comment]
Is this a common error?

anaconda3\lib\site-packages\PIL\TiffImagePlugin.py", line 1182, in _load_libtiff
    raise OSError(err)

**Comment by hmaarrfk** [comment]
can you provide the output of `conda -list`

**Comment by trippingbillies41** [comment]
usage: conda-script.py [-h] [-V] command ...
conda-script.py: error: the following arguments are required: command

**Comment by hmaarrfk** [comment]
sorry, no `dash`. I truely appologize.

```
conda list
```

**Comment by trippingbillies41** [comment]
# packages in environment at /home/morningdew72/anaconda3:
#
# Name                    Version                   Build  Channel
_anaconda_depends         2019.03                  py37_0  
_ipyw_jlab_nb_ext_conf    0.1.0                    py37_0  
_libgcc_mutex             0.1                        main  
alabaster                 0.7.12                   py37_0  
altgraph                  0.16.1                     py_0    conda-forge
anaconda                  custom                   py37_1  
anaconda-client           1.7.2                    py37_0  
anaconda-navigator        1.9.7                    py37_0  
anaconda-project          0.8.4                      py_0  
asn1crypto                1.3.0                    py37_0  
astroid                   2.3.3                    py37_0  
astropy                   4.0              py37h7b6447c_0  
atomicwrites              1.3.0                    py37_1  
attrs                     19.3.0                     py_0  
babel                     2.8.0                      py_0  
backcall                  0.1.0                    py37_0  
backports                 1.0                        py_2  
backports.functools_lru_cache 1.6.1                      py_0  
backports.os              0.1.1                    py37_0  
backports.shutil_get_terminal_size 1.0.0                    py37_2  
backports.tempfile        1.0                        py_1  
backports.weakref         1.0.post1                  py_1  
beautifulsoup4            4.8.2                    py37_0  
bitarray                  1.2.1            py37h7b6447c_0  
bkcharts                  0.2                      py37_0  
blas                      1.0                         mkl  
bleach                    3.1.0                    py37_0  
blosc                     1.16.3               hd408876_0  
bokeh                     1.4.0                    py37_0  
boto                      2.49.0                   py37_0  
bottleneck                1.3.1            py37hdd07704_0  
bzip2                     1.0.8                h7b6447c_0  
ca-certificates           2019.11.28           hecc5488_0    conda-forge
cairo                     1.14.12              h8948797_3  
certifi                   2019.11.28       py37hc8dfbb8_1    conda-forge
cffi                      1.14.0           py37h2e261b9_0  
chardet                   3.0.4                 py37_1003  
click                     7.0                      py37_0  
cloudpickle               1.3.0                      py_0  
clyent                    1.2.2                    py37_1  
colorama                  0.4.3                      py_0  
conda                     4.8.3            py37hc8dfbb8_1    conda-forge
conda-build               3.18.11                  py37_0  
conda-env                 2.6.0                         1  
conda-package-handling    1.6.0            py37h7b6447c_0  
conda-verify              3.4.2                      py_1  
contextlib2               0.6.0.post1                py_0  
cryptography              2.8              py37h1ba5d50_0  
curl                      7.68.0               hbc83047_0  
cycler                    0.10.0                   py37_0  
cython                    0.29.15          py37he6710b0_0  
cytoolz                   0.10.1           py37h7b6447c_0  
dask                      2.10.1                     py_0  
dask-core                 2.10.1                     py_0  
dbus                      1.13.12              h746ee38_0  
decorator                 4.4.1                      py_0  
defusedxml                0.6.0                      py_0  
distributed               2.10.0                     py_0  
docutils                  0.16                     py37_0  
entrypoints               0.3                      py37_0  
et_xmlfile                1.0.1                    py37_0  
expat                     2.2.6                he6710b0_0  
fastcache                 1.1.0            py37h7b6447c_0  
ffmpeg                    4.0.2                ha0c5888_2    conda-forge
fftw                      3.3.8           nompi_h7f3a6c3_1110    conda-forge
filelock                  3.0.12                     py_0  
flake8                    3.7.9                    py37_0    conda-forge
flake8-polyfill           1.0.2                      py_0    conda-forge
flake8-quotes             2.1.1                      py_0    conda-forge
flask                     1.1.1                      py_0  
fontconfig                2.13.0               h9420a91_0  
freeglut                  3.0.0             hf484d3e_1005    conda-forge
freetype                  2.9.1                h8a8886c_1  
fribidi                   1.0.5                h7b6447c_0  
fsspec                    0.6.2                      py_0  
future                    0.18.2                   py37_0  
get_terminal_size         1.0.0                haa9412d_0  
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
gevent                    1.4.0            py37h7b6447c_0  
ghostscript               9.22              hf484d3e_1001    conda-forge
giflib                    5.1.9                h516909a_0    conda-forge
glib                      2.63.1               h5a9c865_0  
glob2                     0.7                        py_0  
gmp                       6.1.2                h6c8ec71_1  
gmpy2                     2.0.8            py37h10f8cd9_2  
gnutls                    3.5.19               h2a4e5f8_1    conda-forge
graphite2                 1.3.13               h23475e2_0  
graphviz                  2.38.0            hcf1ce16_1009    conda-forge
greenlet                  0.4.15           py37h7b6447c_0  
gst-plugins-base          1.14.5               h0935bb2_1    conda-forge
gstreamer                 1.14.5               h36ae1b5_1    conda-forge
h5py                      2.8.0           py37h3010b51_1003    conda-forge
harfbuzz                  1.8.8                hffaf4a1_0  
hdf5                      1.10.2               hc401514_3    conda-forge
heapdict                  1.0.1                      py_0  
html5lib                  1.0.1                    py37_0  
hypothesis                5.4.1                      py_0  
icu                       58.2                 h9c2bf20_1  
idna                      2.8                      py37_0  
imageio                   2.6.1                    py37_0  
imagemagick               7.0.8_11        pl526hc610aec_0    conda-forge
imagesize                 1.2.0                      py_0  
importlib_metadata        1.5.0                    py37_0  
intel-openmp              2020.0                      166  
ipykernel                 5.1.4            py37h39e3cac_0  
ipython                   7.12.0           py37h5ca1d4c_0  
ipython_genutils          0.2.0                    py37_0  
ipywidgets                7.5.1                      py_0  
isort                     4.3.21                   py37_0  
itsdangerous              1.1.0                    py37_0  
jasper                    2.0.14               habb8e15_1    conda-forge
jbig                      2.1                  hdba287a_0  
jdcal                     1.4.1                      py_0  
jedi                      0.16.0                   py37_0  
jeepney                   0.4.2                      py_0  
jinja2                    2.11.1                     py_0  
joblib                    0.14.1                     py_0  
jpeg                      9c                h14c3975_1001    conda-forge
json5                     0.9.1                      py_0  
jsonschema                3.2.0                    py37_0  
jupyter                   1.0.0                    py37_7  
jupyter_client            5.3.4                    py37_0  
jupyter_console           6.1.0                      py_0  
jupyter_core              4.6.1                    py37_0  
jupyterlab                1.2.6              pyhf63ae98_0  
jupyterlab_server         1.0.6                      py_0  
keyring                   21.1.0                   py37_0  
kivy                      1.11.1           py37h2394fcc_0    conda-forge
kiwisolver                1.1.0            py37he6710b0_0  
krb5                      1.17.1               h173b8e3_0  
lazy-object-proxy         1.4.3            py37h7b6447c_0  
ld_impl_linux-64          2.33.1               h53a641e_7  
libarchive                3.3.3                h5d8350f_5  
libcurl                   7.68.0               h20c2e04_0  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran               3.0.0                         1    conda-forge
libgfortran-ng            7.3.0                hdf63c60_0  
libglu                    9.0.0             hf484d3e_1000    conda-forge
libiconv                  1.15              h516909a_1005    conda-forge
libjpeg-turbo             2.0.3                h516909a_1    conda-forge
liblief                   0.9.0                h7725739_2  
libopencv                 3.4.2                hb342d67_1  
libpng                    1.6.37               hbc83047_0  
libsodium                 1.0.16               h1bed415_0  
libssh2                   1.8.2                h1ba5d50_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.1.0                h2733197_0  
libtool                   2.4.6                h7b6447c_5  
libuuid                   1.0.3                h1bed415_2  
libwebp                   0.5.2                         7    conda-forge
libxcb                    1.13                 h1bed415_1  
libxml2                   2.9.9                hea5a465_1  
libxslt                   1.1.33               h7d1a2b0_0  
llvmlite                  0.31.0           py37hd408876_0  
locket                    0.2.0                    py37_1  
lxml                      4.5.0            py37hefd8a0e_0  
lz4-c                     1.8.1.2              h14c3975_0  
lzo                       2.10                 h49e0be7_2  
macholib                  1.11                       py_0    conda-forge
markupsafe                1.1.1            py37h7b6447c_0  
matplotlib                3.1.3                    py37_0  
matplotlib-base           3.1.3            py37hef1b27d_0  
mccabe                    0.6.1                    py37_1  
mistune                   0.8.4            py37h7b6447c_0  
mkl                       2020.0                      166  
mkl-service               2.3.0            py37he904b0f_0  
mkl_fft                   1.0.15           py37ha843d7b_0  
mkl_random                1.1.0            py37hd6b4f25_0  
mock                      4.0.1                      py_0  
more-itertools            8.2.0                      py_0  
mpc                       1.1.0                h10f8cd9_1  
mpfr                      4.0.1                hdf1c602_3  
mpmath                    1.1.0                    py37_0  
msgpack-python            0.6.1            py37hfd86e86_1  
multipledispatch          0.6.0                    py37_0  
navigator-updater         0.2.1                    py37_0  
nbconvert                 5.6.1                    py37_0  
nbformat                  5.0.4                      py_0  
ncurses                   6.1                  he6710b0_1  
nettle                    3.3                           0    conda-forge
networkx                  2.4                        py_0  
nltk                      3.4.5                    py37_0  
nose                      1.3.7                    py37_2  
notebook                  6.0.3                    py37_0  
numba                     0.48.0           py37h0573a6f_0  
numexpr                   2.7.1            py37h423224d_0  
numpy                     1.18.1           py37h4f9e942_0  
numpy-base                1.18.1           py37hde5b4d6_1  
numpydoc                  0.9.2                      py_0  
olefile                   0.46                     py37_0  
opencv                    3.4.2            py37h6fd60c2_1  
openh264                  1.8.0             hdbcaa40_1000    conda-forge
openjpeg                  2.3.1                h981e76c_3    conda-forge
openpyxl                  3.0.3                      py_0  
openssl                   1.1.1e               h516909a_0    conda-forge
packaging                 20.1                       py_0  
pandas                    1.0.1            py37h0573a6f_0  
pandoc                    2.2.3.2                       0  
pandocfilters             1.4.2                    py37_1  
pango                     1.40.14              he752989_2    conda-forge
parso                     0.6.1                      py_0  
partd                     1.1.0                      py_0  
patchelf                  0.10                 he6710b0_0  
path                      13.1.0                   py37_0  
path.py                   12.4.0                        0  
pathlib2                  2.3.5                    py37_0  
patsy                     0.5.1                    py37_0  
pcre                      8.43                 he6710b0_0  
pdf2image                 1.12.1                     py_0    conda-forge
pep8                      1.7.1                    py37_0  
pep8-naming               0.9.1                      py_0    conda-forge
perl                      5.26.2            h516909a_1006    conda-forge
pexpect                   4.8.0                    py37_0  
pickleshare               0.7.5                    py37_0  
pillow                    7.0.0            py37hb39fc2d_0  
pip                       20.0.2                   py37_1  
pixman                    0.38.0               h7b6447c_0  
pkg-config                0.29.2            h516909a_1006    conda-forge
pkginfo                   1.5.0.1                  py37_0  
pluggy                    0.13.1                   py37_0  
ply                       3.11                     py37_0  
prettytable               0.7.2                      py_3    conda-forge
prometheus_client         0.7.1                      py_0  
prompt_toolkit            3.0.3                      py_0  
psutil                    5.6.7            py37h7b6447c_0  
ptyprocess                0.6.0                    py37_0  
py                        1.8.1                      py_0  
py-lief                   0.9.0            py37h7725739_2  
py-opencv                 3.4.2            py37hb342d67_1  
pycodestyle               2.5.0                    py37_0  
pycosat                   0.6.3            py37h7b6447c_0  
pycparser                 2.19                     py37_0  
pycrypto                  2.6.1            py37h14c3975_9  
pycurl                    7.43.0.5         py37h1ba5d50_0  
pyflakes                  2.1.1                    py37_0  
pygments                  2.5.2                      py_0  
pyinstaller               3.6              py37hed695b0_0    conda-forge
pylint                    2.4.4                    py37_0  
pyodbc                    4.0.30           py37he6710b0_0  
pyopenssl                 19.1.0                   py37_0  
pyparsing                 2.4.6                      py_0  
pyqt                      5.9.2            py37h05f1152_2  
pyrsistent                0.15.7           py37h7b6447c_0  
pysocks                   1.7.1                    py37_0  
pytables                  3.4.4            py37ha205bf6_0  
pytesseract               0.3.2                      py_0    conda-forge
pytest                    5.3.5                    py37_0  
pytest-arraydiff          0.3              py37h39e3cac_0  
pytest-astropy            0.7.0                      py_0  
pytest-astropy-header     0.1.2                      py_0  
pytest-doctestplus        0.5.0                      py_0  
pytest-openfiles          0.4.0                      py_0  
pytest-remotedata         0.3.2                    py37_0  
python                    3.7.6                h0371630_2  
python-dateutil           2.8.1                      py_0  
python-libarchive-c       2.8                     py37_13  
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2019.3                     py_0  
pywavelets                1.1.1            py37h7b6447c_0  
pyyaml                    5.3              py37h7b6447c_0  
pyzbar                    0.1.5                    py37_0    lightsource2-tag
pyzmq                     18.1.1           py37he6710b0_0  
qrcode                    6.1                        py_0    conda-forge
qt                        5.9.7                h5867ecd_1  
qtawesome                 0.6.1                      py_0  
qtconsole                 4.6.0                      py_1  
qtpy                      1.9.0                      py_0  
readline                  7.0                  h7b6447c_5  
requests                  2.22.0                   py37_1  
ripgrep                   11.0.2               he32d670_0  
rope                      0.16.0                     py_0  
ruamel_yaml               0.15.87          py37h7b6447c_0  
scikit-image              0.16.2           py37h0573a6f_0  
scikit-learn              0.22.1           py37hd81dba3_0  
scipy                     1.4.1            py37h0b6359f_0  
sdl2                      2.0.10               he1b5a44_0    conda-forge
sdl2_image                2.0.5                h0e9dbb3_0    conda-forge
sdl2_mixer                2.0.4                he1b5a44_0    conda-forge
sdl2_ttf                  2.0.15               h7d98c4e_0    conda-forge
seaborn                   0.10.0                     py_0  
secretstorage             3.1.2                    py37_0  
send2trash                1.5.0                    py37_0  
setuptools                45.2.0                   py37_0  
simplegeneric             0.8.1                    py37_2  
singledispatch            3.4.0.3                  py37_0  
sip                       4.19.8           py37hf484d3e_0  
six                       1.14.0                   py37_0  
smpeg2                    2.0.0                hfc679d8_1    conda-forge
snappy                    1.1.7                hbae5bb6_3  
snowballstemmer           2.0.0                      py_0  
sortedcollections         1.1.2                    py37_0  
sortedcontainers          2.1.0                    py37_0  
soupsieve                 1.9.5                    py37_0  
sphinx                    2.4.0                      py_0  
sphinxcontrib             1.0                      py37_1  
sphinxcontrib-applehelp   1.0.1                      py_0  
sphinxcontrib-devhelp     1.0.1                      py_0  
sphinxcontrib-htmlhelp    1.0.2                      py_0  
sphinxcontrib-jsmath      1.0.1                      py_0  
sphinxcontrib-qthelp      1.0.2                      py_0  
sphinxcontrib-serializinghtml 1.1.3                      py_0  
sphinxcontrib-websupport  1.2.0                      py_0  
spyder                    3.3.6                    py37_0  
spyder-kernels            0.5.2                    py37_0  
sqlalchemy                1.3.13           py37h7b6447c_0  
sqlite                    3.31.1               h7b6447c_0  
statsmodels               0.11.0           py37h7b6447c_0  
sympy                     1.5.1                    py37_0  
tbb                       2020.0               hfd86e86_0  
tblib                     1.6.0                      py_0  
terminado                 0.8.3                    py37_0  
testpath                  0.4.4                      py_0  
texttable                 1.6.2                      py_0    conda-forge
tk                        8.6.8                hbc83047_0  
toolz                     0.10.0                     py_0  
tornado                   6.0.3            py37h7b6447c_3  
tqdm                      4.42.1                     py_0  
traitlets                 4.3.3                    py37_0  
unicodecsv                0.14.1                   py37_0  
unixodbc                  2.3.7                h14c3975_0  
urllib3                   1.25.8                   py37_0  
wand                      0.5.6                      py_0    conda-forge
wcwidth                   0.1.8                      py_0  
webencodings              0.5.1                    py37_1  
werkzeug                  1.0.0                      py_0  
wheel                     0.34.2                   py37_0  
widgetsnbextension        3.5.1                    py37_0  
wrapt                     1.11.2           py37h7b6447c_0  
wurlitzer                 2.0.0                    py37_0  
x264                      1!152.20180806       h14c3975_0    conda-forge
xlrd                      1.2.0                    py37_0  
xlsxwriter                1.2.7                      py_0  
xlwt                      1.3.0                    py37_0  
xorg-fixesproto           5.0               h14c3975_1002    conda-forge
xorg-inputproto           2.3.2             h14c3975_1002    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.2                h470a237_5    conda-forge
xorg-libx11               1.6.9                h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxfixes            5.0.3             h516909a_1004    conda-forge
xorg-libxi                1.7.10               h516909a_0    conda-forge
xorg-libxpm               3.5.13               h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-libxt                1.1.5             h516909a_1003    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.4                h14c3975_4  
yaml                      0.1.7                had09818_2  
zbar                      0.10              hf484d3e_1000    conda-forge
zeromq                    4.3.1                he6710b0_3  
zict                      1.0.0                      py_0  
zipp                      2.2.0                      py_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.3.7                h0b5b093_0

**Comment by hmaarrfk** [comment]
it seems to me that you are using linux. Since the original issue depends on Windows, could you please open a new issue so that them maintainers of this repo can help you get going.

I don't htink you issue is related to this one.

**Comment by trippingbillies41** [comment]
Sorry, I just turned on my PC and responded. Forgot that it was about a Windows issue. I edited my comment to reflex the conda list from my Windows partition.

**Comment by hmaarrfk** [comment]
so there are a few things going on in your system:
1. You are using base. You shouldn't use base, especially with conda-forge. Please attempt to create an isolated environment for your dependencies.


I Think until the motions are going through upstream, and finally trickle down, this issue won't get fixed easily, and we will have to wait for new releases of libtiff. That is likely the easiest way.

Maybe use tifffile instead?

**Comment by trippingbillies41** [comment]
I guess I still don't understand the purpose of environments. Do I make a new one for each project? I just thought adding everything to base would allow me pull any module I wanted.

**Comment by hmaarrfk** [comment]
generally, you want to have one for each project.
The issue you are facing is:
1. A bug in the C world.
2. The fact that the default packages you installed are not always compatible with conda-forge (especially problematic for C packages).

In either case, this isn't the right issue to discuss this (discussing environment was not part of the original issue). Do open an other one here, or probably on stack overflow or some other Q&A forum https://forum.image.sc/

**Comment by trippingbillies41** [comment]
Thanks, I'll learn about those issues!

**Comment by scimax** [comment]
I just installed the patched version and for the moment it seems to work. Thanks, @hmaarrfk !

**Comment by nkoerb** [comment]
I have the same issue on windows server 2016 with pillow 7.1.2, libtif 4.1 and conda, when I downgrade to Pillow 6.2.1 and libtiff 4.0.10 the code works.
```
from PIL import Image
image = Image.open('mytiff.tif')
image.show()
```
OSError -2

**Comment by kmilos** [comment]
@nkoerb Only libtiff 4.1.0 build h885aae3_4 or later from conda-forge channel works, not from defaults

**Comment by nkoerb** [comment]
@kmilos thanks for the info

**Comment by chibbi30j1** [comment]
How to install the patched version and uninstall the current one?

**Comment by shaibagon** [comment]
@chibbi30j1 
This worked for me:
```
conda install libtiff=4.1.0=h885aae3_4 -c conda-forge
```

**Comment by radarhere** [comment]
@kmilos is this resolved now, thanks to #5359?

**Comment by kmilos** [comment]
I believe so, as far as ootb Pillow wheels are concerned.

@hmaarrfk conda-forge will have to re-adjust again though (either turn off Win32 I/O in Pillow > 8.2.0 with a new patch, or re-enable Win32 I/O for libitff, Pillow, and poppler...)
